pkg/operator/operator.go

```diff
@@ -32,12 +32,14 @@ func New(metricsAddr string) (*Operator, error) {
 		return nil, fmt.Errorf("unable to create manager: %w", err)
 	}
 
-	if err := stackctrl.RegisterWithManager(mgr, stackctrl.Options{InstanceSelector: instanceSelector}); err != nil {
+	if err := stackctrl.RegisterWithManager(mgr, stackctrl.Options{InstanceSelector: instanceSelector, EnableGrafanaOperator: enableGrafanaOperator}); err != nil {
 		return nil, fmt.Errorf("unable to register monitoring stack controller: %w", err)
 	}
 
-	if err := goctrl.RegisterWithManager(mgr); err != nil {
-		return nil, fmt.Errorf("unable to register the grafana operator controller with the manager: %w", err)
+	if enableGrafanaOperator {
```

Comment:
if granfana-operator isn't enabled, IMO, we should delete any existing deployment, which IIUC, this wouldn't, thus the operator could have a dangling grafana deployed.